static/app/components/tabs/index.tsx

```diff
@@ -0,0 +1,69 @@
+import 'intersection-observer'; // polyfill
+
+import {createContext, useState} from 'react';
+import styled from '@emotion/styled';
+import {AriaTabListProps} from '@react-aria/tabs';
+import {Item} from '@react-stately/collections';
+import {TabListProps, TabListState} from '@react-stately/tabs';
+import {ItemProps, Orientation} from '@react-types/shared';
+
+import {TabList} from './tabList';
+import {TabPanels} from './tabPanels';
+
+const _Item = Item as <T>(props: ItemProps<T> & {disabled?: boolean}) => JSX.Element;
+export {_Item as Item, TabList, TabPanels};
+
+export interface TabsProps<T> extends TabListProps<T>, AriaTabListProps<T> {
+  className?: string;
+  disabled?: boolean;
+}
+
+interface TabContext<T> {
+  rootProps: TabsProps<T> & {orientation: Orientation};
+  setTabListState: (state: TabListState<T>) => void;
+  tabListState?: TabListState<T>;
```

Comment:
shall this prop be required?

Reply:
` tabListState` is initially undefined, until it's loaded in `TabsList`. I provided defaults for `rootProps` and `setTabListState`, but not `tabListState` because it's a rather complex object.